awx/ui/client/src/templates/survey-maker/surveys/init.factory.js

```diff
@@ -382,9 +382,9 @@ export default
                         data.default = scope.default_password;
                     } else if (scope.type.type === 'multiselect') {
                           data.default = scope.default_multiselect;
-                    } else if (scope.type.type === 'float') {
+                    } else if (scope.type.type === 'float' && scope.default_float) {
                         data.default = scope.default_float;
-                    } else if (scope.type.type === 'integer') {
+                    } else if (scope.type.type === 'integer' && scope.default_int) {
```

Comment:
@saito-hideki 
We can't rely on the truthiness of `scope.default_int` for all cases. For example, this would prevent someone from updating the default value to `0`.

Reply:
@saito-hideki 
When cleared this can become an empty string or null depending on how it's been initialized.

```js
...
} else if (scope.type.type === 'float' &&
    scope.default_float !== '' &&
    scope.default_float !== null) {
    data.default = scope.default_float;
} else if (scope.type.type === 'integer' &&
    scope.default_int !== '' &&
    scope.default_int !== null) {
    data.default = scope.default_int;
} else {
...
```
Something like the above _might_ be all you need here - please give this a look and let me know what you find.

Reply:
@jakemcdermott Thank you for your kind review!
As you pointed out, there was a problem with this condition handling. So I have fixed it.